### ISSUE TITLE
Required back-reference with named "ambiguous" relation

### ISSUE THREAD

**Issue by LFP6** [graphcool/prisma#1569]
**Current behavior**

When using an `@relation` with a name, no back-reference, and an otherwise ambiguous relation, it fails with the error "A relation directive with a name must appear exactly 2 times."

**Reproduction**
```
type Post {
  id: ID! @unique
  comment: Comment! @relation(name: "Comment1")
  otherComment: Comment! @relation(name: "Comment2")
}

type Comment {
  id: ID! @unique
}
```

**Expected behavior?**
Per my comments in #1380,  #1489 / 315d880 was merged with updates to make `@relation` and back-references optional. According to the spec in https://github.com/graphcool/graphcool/issues/1160#issuecomment-345373277 and other (older) issues, back-references should still be optional with named relations. It was my understanding that the back-reference should be optional period (ie, even though it's ambiguous I still don't want it exposed on the other type), so I'm guessing that either there's still something about the handling of ambiguous relations that requires this (and hopefully can be adjusted to not be this way), or the validation was just not removed when it should have been.

**Comment by emmenko** [comment]
I'm also trying out the current beta and run into the same issue:

```graphql
type TeamMatch {
  key: String! @unique
}

type Match {
  teamLeft: TeamMatch
  teamRight: TeamMatch
  winner: TeamMatch
}
```

With errors:
```
  Match
    ✖ The relation field `teamLeft` must specify a `@relation` directive: `@relation(name: "MyRelation")`
    ✖ The relation field `teamRight` must specify a `@relation` directive: `@relation(name: "MyRelation")`
    ✖ The relation field `winner` must specify a `@relation` directive: `@relation(name: "MyRelation")`
```

If I provide a `match` field in the `TeamMatch` type I also get an error:

```graphql
type TeamMatch {
  key: String! @unique
  match: Match!
}

type Match {
  teamLeft: TeamMatch
  teamRight: TeamMatch
  winner: TeamMatch
}
```

```
  Match
    ✖ The relation field `teamLeft` must specify a `@relation` directive: `@relation(name: "MyRelation")`
    ✖ The relation field `teamRight` must specify a `@relation` directive: `@relation(name: "MyRelation")`
    ✖ The relation field `winner` must specify a `@relation` directive: `@relation(name: "MyRelation")`
```

And if I define some relations I also get an error:

```graphql
type TeamMatch {
  key: String! @unique
  match: Match!
}

type Match {
  teamLeft: TeamMatch @relation(name: "TeamMatchLeft")
  teamRight: TeamMatch @relation(name: "TeamMatchRight")
  winner: TeamMatch @relation(name: "TeamMatchWinner")
}
```

```
  Match
    ✖ A relation directive with a name must appear exactly 2 times. Relation name: 'TeamMatchLeft'
    ✖ A relation directive with a name must appear exactly 2 times. Relation name: 'TeamMatchRight'
    ✖ A relation directive with a name must appear exactly 2 times. Relation name: 'TeamMatchWinner'
```

---
Am I doing something wrong or is there an actual problem?

**Comment by marktani** [comment]
Does this still happen on the latest version?

**Comment by doodzik** [comment]
I get the same error with the latest version @marktani

**Comment by ipodishima** [comment]
I also get the same error. I imagine that for now you need to explicitely name the references on both side?

**Comment by mburak** [comment]
Hi guys, Is this already fixed? I tried the latest version (1.8.3) and I'm still getting the same issue.

**Comment by emmenko** [comment]
I just tried without the back references and is working for me 🙌 (`2368 deletions(-)` 🤩🙏)

```bash
$ git diff --stat src/generated/prisma.graphql
 src/generated/prisma.graphql | 2583 ++---------------------
 1 file changed, 215 insertions(+), 2368 deletions(-)
```
